pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferenceManagerWorkerService.scala

```diff
@@ -3,31 +3,43 @@ package uk.ac.wellcome.platform.inference_manager.services
 import akka.Done
 import akka.http.scaladsl.model.HttpResponse
 import akka.actor.ActorSystem
-import akka.stream.scaladsl.{FlowWithContext, Source}
+import akka.stream.scaladsl.{Flow, FlowWithContext, Source}
 import grizzled.slf4j.Logging
 import software.amazon.awssdk.services.sqs.model.Message
 import uk.ac.wellcome.bigmessaging.message.BigMessageStream
 import uk.ac.wellcome.messaging.MessageSender
-import uk.ac.wellcome.models.work.internal.AugmentedImage
+import uk.ac.wellcome.models.work.internal.{AugmentedImage, InferredData}
 import uk.ac.wellcome.models.Implicits._
+import uk.ac.wellcome.platform.inference_manager.adapters.{
+  InferrerAdapter,
+  InferrerResponse
+}
 import uk.ac.wellcome.platform.inference_manager.models.DownloadedImage
 import uk.ac.wellcome.typesafe.Runnable
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
 import scala.util.{Failure, Success, Try}
 
+case class AdapterResponseBundle[ImageType](
+  image: ImageType,
+  adapter: InferrerAdapter,
+  response: Try[InferrerResponse]
```

Comment:
Related to my comment above, there is nothing you can do with the success type of a `Try[InferrerResponse]` as the trait does provide any interface. Are you planning to add some required functionality to the trait so the response can be used? If not I see 2 options:

* Making this an error instead (e.g. `error: Option[Exception]` or something)
* Adding another type parameter to the case class so we can use the concrete type: `AdapterResponseBundle[ImageType, Response]`